### ISSUE TITLE
*Question* Depth API Supported Devices

### ISSUE THREAD

**Issue by ValentijnSlijper** [google-ar/arcore-android-sdk#1124]
**There is no rush to answer these questions.**

I am quite curious about what information you actually need to add a new phone to DepthAPI's supported list. 

- Do you only need information about the camera lens or also the hardware of the device?
- How long will it take for you to add a new device to this list?
- How do you choose which device you are going to support?

Thanks for time to answer these questions in advance.

(Doing some research for an internship. The device I am currently using Xiaomi Mi Mix 3)

**Comment by boehm-e** [comment]
From what I understand reading [this](https://developers.googleblog.com/2019/12/blending-realities-with-arcore-depth-api.html), the Depth API, is just comparing subsequent pictures from your phone (so at different locations) to generate a depth map, like our eyes do. So I don't see why they would any other informations about the phone, since they already have a calibration file containing all relevant informations for each phone supporting ARCore.
Even their own phones are not all compatible with the depth API...

**Comment by ValentijnSlijper** [comment]
@boehm-e 

What is exactly why I am curious about this. If they only are using 2D images to create a depth map why are not all phones supported. And if this is not the case what are the limitations then? The hardware in the phone?

**Comment by boehm-e** [comment]
@ValentijnSlijper 

Yes it could be hardware, but my honor 10 does support depth API, and it is really not high-end phone, it struggles and over-heat when I use the depth API, but it does work (kind of).
Any snapdragon phone would handle it better i guess. You say you have a mi mix 3 ? It has a snapdragon 845, as my regretted pocophone f1 had, and the pocophone is compatible with the depth API. So I see no reason why your phone or any other phones could not get depth API, maybe this is a marketing issue, or some other stupid thing --'
And, now that many phones (including the 200$ poco x3) have Tof sensor, I dont know when ARCore will use that. I don't event know if the ToF data are accessible... at least Apple have a great SDK, with a lot of support and are being very clear about all of that.

**Comment by ValentijnSlijper** [comment]
@boehm-e 
You are spelling out my thoughts because I was wondering the same thing. There are many phones on the list that have worse specs then my device and they do support DepthAPI. I'm not saying I have a super High-End phone but it should certainly be good enough for DepthAPI which is what I think anyway.

Like you, I thought maybe it had something to do with marking. That's why I hoped to get some answers in a subtle way :)

And I also hoped that they would support several devices in the future, including mine of course. 
So that I can continue my project for an internship.

Anyways thanks you for your insight.

**Comment by epadam** [comment]
https://medium.com/6d-ai/why-is-arkit-better-than-the-alternatives-af8871889d6a
This article explains very well what hardware AR requires and some other details.
I think some phones are not supported because their IMU is not calibrated with Google, which needs extra cost.
